tremor-pipeline/src/op/trickle/select.rs

```diff
@@ -646,221 +648,647 @@ impl Operator for TrickleSelect {
+                    set_window(consts, Value::from(window.name.to_string()))?;
+                    // This is sound since we only add mutability to groups
+                    let this_groups = &mut window.dims.groups;
+                    let last_groups = &mut window.last_dims.groups;
+                    let window_impl = &window.window_impl;
                     let mut idgen = self.event_id_gen;
-                    let (_, prev_group) = prev_groups
+                    let (_, this_group) = this_groups
                         .raw_entry_mut()
                         .from_key(&group_str)
                         .or_insert_with(|| {
+                            // This is sound since we only add mutability to groups
                             (
                                 group_str.clone(),
                                 last_groups.remove(&group_str).unwrap_or_else(|| GroupData {
                                     window: window_impl.clone(),
                                     aggrs: aggregates.clone(),
                                     group: group_value.clone_static(),
-                                    id: idgen.next_id(),
+                                    id: idgen.next_id(), // after all this is a new event
                                 }),
                             )
                         });
-                    for (aggr, this) in prev_group.aggrs.iter().zip(this_group.aggrs.iter_mut()) {
-                        this.invocable.merge(&aggr.invocable).map_err(|e| {
+                    let window_event = this_group.window.on_event(&event)?;
+                    if window_event.emit && !window_event.include {
+                        // push
+                        let env = Env {
+                            context: &ctx,
+                            consts: &consts,
+                            aggrs: &this_group.aggrs,
+                            meta: &node_meta,
+                            recursion_limit: tremor_script::recursion_limit(),
+                        };
+                        let result = stmt.target.run(
+                            opts,
+                            &env,
+                            unwind_event,
+                            state,
+                            event_meta,
+                            &local_stack,
+                        )?;
+
+                        if let Some(guard) = &stmt.maybe_having {
+                            let test =
+                                guard.run(opts, &env, &result, state, &NULL, &local_stack)?;
+                            if let Some(test) = test.as_bool() {
+                                if !test {
+                                    continue;
+                                }
+                            } else {
+                                let s: &Select = &stmt;
+                                return tremor_script::errors::query_guard_not_bool(
```

Comment:
could we add a query_guard_not_bool_err function that returns the Error instead of Result to avoid return ...?;?

Reply:
we need the `?` as the error types differ i think.

Reply:
`.into()` should do the same and hopefully be more readable.